prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,194 @@
+# DELETED: comm_type, LOAD, vmtsc
+
+import mileage
+import charging
+import numpy as np
+
+
+def immediate_charging(
+    census_region, model_year, veh_range, kwhmi, power, location_strategy
+):
+    """Immediate charging function
+
+    :param int census_region: any of the 9 census regions defined by US census bureau.
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040, 2050.
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int kwhmi: fuel efficiency, should vary based on vehicle type and model_year.
+    :param int power: charger power, EVSE kW.
+    :param int location_strategy: where the vehicle can charge-1, 2, or 3; 1 = home only, 2 = home and work, 3 = everywhere
+    :return: (*numpy.ndarray*) -- charging profiles.
+    """
+    # load NHTS data from function
+    newdata = mileage.remove_ldt(mileage.load_data(census_region))
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata.insert(28, "trip start battery charge", [0] * len(newdata))
+    newdata.insert(29, "trip end battery charge", [0] * len(newdata))
+    newdata.insert(30, "charging power", [0] * len(newdata))
+    newdata.insert(31, "charging time", [0] * len(newdata))
+    # jx: never actually used... delete?
+    newdata.insert(32, "charging consumption", [0] * len(newdata))  # needs double check
+    newdata.insert(33, "BEV could be used", [0] * len(newdata))
+    newdata.insert(34, "trip number", [0] * len(newdata))

```

Comment:
If all we're doing is initializing new columns full of zeros, we can do simply: `newdata["trip start battery charge"] = 0`, or better yet we can avoid pre-initializing at all and just add the data to the data frame once it's calculated.

Reply:
Fixed